deepspeed/pt/deepspeed_light.py

```diff
@@ -879,22 +885,22 @@ def csr_all_gather(self, value):
         if value.dim() == 1:
             if fill_size > 0:
                 value = torch.cat([value, value.new_zeros(fill_size)])
-            tensor_list = [
-                value.new_zeros(max_size) for _ in range(dist.get_world_size())
-            ]
+            tensor_list = [value.new_zeros(max_size) for _ in range(self.dp_world_size)]
```

Comment:
I made an educated guess that this should be done over the data-parallel instead of global world. @jeffra could you please confirm?